src/arrayMethodJoin.js

```diff
@@ -4,8 +4,24 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+    let firstLoop = true;
+
+    for (let string of this) {
+      if (string === undefined || string === null) {
+        string = '';
+      }
+
+      if (firstLoop) {
+        result = String(string);
+        firstLoop = false;
+      } else {
+        result = `${result}${separator}${string}`;
```

Comment:
won't this do the thing?
```suggestion
        result += `${separator}${string}`;
```